File: 1. Two Sum/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target) {
  int _[100001] = {0}, *index_plus_one = _ + 50000;
  for (int i = 0; i < numsSize; i++) {
    int rest = target - nums[i];
    if (index_plus_one[rest]) {
      int *ans = malloc(sizeof(int) * 2);
      ans[0] = i;
      ans[1] = index_plus_one[rest] - 1;
      return ans;
    }
    else
      index_plus_one[nums[i]] = i + 1;
  }
  return NULL;
}

File: 1. Two Sum/solution.go
func twoSum(nums []int, target int) []int {
	lookup := make(map[int]int)
	for i, num := range nums {
		if j, ok := lookup[target-num]; ok {
			return []int{j, i}
		}
		lookup[nums[i]] = i
	}
	return nil
}

/* 
 * 解題思維：
 * 使用 map 將所有組合比對
 * 如果比對到一組就馬上回傳
 */